ethcore/src/evm/interpreter/memory.rs

```diff
@@ -109,6 +114,20 @@ impl Memory for Vec<u8> {
 			Memory::resize(self, size)
 		}
 	}
+
+	fn into_return_data(mut self, offset: U256, size: U256) -> ReturnData {
+		let offset = offset.low_u64() as usize;
+		let size = size.low_u64() as usize;
+		if !is_valid_range(offset, size) {
```

Comment:
Memory is guaranteed to be allocated at this point. This is assured by mem/gas requirements checking before each instruction. That's why none of the methods in this struct check for bounds.